### PR TITLE
Update the version of `DynamoDb`.

### PR DESCRIPTION
**Pull request by MathieuDutSik**
## Motivation

The version of `DynamoDb` that we used was pretty ancient and this could have been part of the problem we experienced.
Since anyway, we have no reason to stick to the current version, it made sense to upgrade.

## Proposal

The upgrade was done at the level of the `Cargo.toml` of course, but there was plenty of other changes to do since the code had been significantly reorganized.

## Test Plan

The CI should take care of that.

## Release Plan

Nothing to be done.

## Links

- [reviewer checklist](https://github.com/linera-io/linera-protocol/blob/main/CONTRIBUTING.md#reviewer-checklist)

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -33,11 +33,11 @@ async-graphql-axum = "5.0.10"
 async-lock = "2.8.0"
 async-stream = "0.3.5"
 async-trait = "0.1.74"
-aws-config = "0.14.0"
-aws-sdk-dynamodb = "0.14.0"
-aws-sdk-s3 = "0.14.0"
-aws-smithy-http = "0.44.0"
-aws-types = "0.14.0"
+aws-config = "0.56.1"
+aws-sdk-dynamodb = "0.34.0"
```
Comment (ma2bd) [inline Cargo.toml:37]: I believe `0.35.0` is the last version: https://crates.io/crates/aws-sdk-dynamodb/versions

File: linera-views/src/dynamo_db.rs [hunk 2]
```diff
@@ -7,43 +7,128 @@ use crate::{
...
 };
 use bcs::serialized_size;
 use futures::future::join_all;
 use serde::{Deserialize, Serialize};
 use static_assertions as sa;
-use std::{collections::HashMap, mem, str::FromStr, sync::Arc};
+use std::{collections::HashMap, env, mem, str::FromStr, sync::Arc};
 use thiserror::Error;
 
+use http::uri::InvalidUri;
```
Comment (ma2bd) [inline linera-views/src/dynamo_db.rs:40]: The newline is misplaced.

File: linera-views/src/dynamo_db.rs [hunk 3]
```diff
@@ -7,43 +7,128 @@ use crate::{
...
-    anyhow::{Context, Error},
-    aws_sdk_s3::Endpoint,
-    aws_types::SdkConfig,
-    std::env,
+    anyhow::Error,
     tokio::sync::{Mutex, MutexGuard},
 };
 
+/// Name of the environment variable with the address to a LocalStack instance.
+pub const LOCALSTACK_ENDPOINT: &str = "LOCALSTACK_ENDPOINT";
```
Comment (ma2bd) [inline linera-views/src/dynamo_db.rs:50]: Does it need to be `pub` ?

File: linera-views/src/dynamo_db.rs [hunk 4]
```diff
@@ -568,14 +653,28 @@ impl DynamoDbClientInternal {
...
-            SdkError::ServiceError { err, raw: _ } => match &err.kind {
-                GetItemErrorKind::ResourceNotFoundException(inner_error) => {
-                    inner_error.message
+            SdkError::ServiceError(error) => match error.err() {
+                GetItemError::ResourceNotFoundException(error) => {
+                    println!("2 : error={:?}", error);
+                    if error.message
                         == Some("Cannot do operations on a non-existent table".to_string())
+                    {
+                        println!("Exit case 1");
```
Comment (ma2bd) [inline linera-views/src/dynamo_db.rs:662]: What is this?

**Review by ma2bd — COMMENTED** [review]
Can you re-arrange the commits to be distinct logical changes (and formatted) ?

File: Cargo.toml (hunk 1 above)
Comment (MathieuDutSik) [reply Cargo.toml:37]: Ok, it was after I started working on the problem.

Comment (MathieuDutSik) [reply Cargo.toml:37]: And that turns out to be quite as hard. Will try.

Comment (MathieuDutSik) [reply Cargo.toml:37]: Done.

File: linera-views/src/dynamo_db.rs (hunk 2 above)
Comment (MathieuDutSik) [reply linera-views/src/dynamo_db.rs:40]: Actually, the `InvalidUri` could be eliminated completely.

File: linera-views/src/dynamo_db.rs (hunk 4 above)
Comment (MathieuDutSik) [reply linera-views/src/dynamo_db.rs:662]: Some oversight.

File: linera-views/src/dynamo_db.rs (hunk 3 above)
Comment (MathieuDutSik) [reply linera-views/src/dynamo_db.rs:50]: Actually no.

File: Cargo.toml [hunk 5]
```diff
@@ -33,11 +33,11 @@ async-graphql-axum = "5.0.10"
 async-lock = "2.8.0"
```
Comment (ma2bd) [inline Cargo.toml:33]: I'm assuming that the changes here belong to the next commit

File: linera-views/src/dynamo_db.rs (hunk 4 above)
Comment (ma2bd) [reply linera-views/src/dynamo_db.rs]: Even in tests, log messages need to be meaningful and use the `tracing` crate (and not `print!`).
Can't we log the error types returned by DynamoDb?

Comment (ma2bd) [reply linera-views/src/dynamo_db.rs]: Alright, it looks like this is removed in later commits.

**Review by ma2bd — APPROVED** [review]

**Pull request merged**